### ISSUE TITLE
Build fails: The method 'AssetThumbImageProvider.load' has fewer positional arguments than those of overridden method 'ImageProvider.load'.

### ISSUE THREAD

**Issue by technolion** [Sh1d0w/multi_image_picker#283]
Build failed after upgrading to flutter Channel master, v1.10.15-pre.121. The Error is:

```
The method 'AssetThumbImageProvider.load' has fewer positional arguments than those of overridden method 'ImageProvider.load'.
```

`multi_image_picker` version: 4.5.8

Is it happening on Android: I don't know

Is it happening on iOS: Yes

```
porcupine: tom$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v1.10.15-pre.121, on Mac OS X 10.14.6 18G103, locale de-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
[✓] Chrome - develop for the web
[!] Android Studio (version 3.4)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] VS Code (version 1.39.0)
[✓] Connected device (4 available)

! Doctor found issues in 1 category.
```

**Describe the bug**

The build of multi_image_picker fails with the following output:

```
Xcode build done.                                            8,7s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳

    Compiler message:
    /usr/local/flutter/.pub-cache/hosted/pub.dartlang.org/multi_image_picker-4.5.8/lib/src/asset_thumb_provider.dart:31:24: Error: The method 'AssetThumbImageProvider.load' has fewer positional
    arguments than those of overridden method 'ImageProvider.load'.
      ImageStreamCompleter load(AssetThumbImageProvider key) {
                           ^
    /usr/local/flutter/packages/flutter/lib/src/painting/image_provider.dart:403:24: Context: This is the overridden method ('load').
      ImageStreamCompleter load(T key, DecoderCallback decode);
                           ^
    Exception: Errors during snapshot creation: null
    #0      KernelSnapshot.build (package:flutter_tools/src/build_system/targets/dart.dart:225:7)
    <asynchronous suspension>
    #1      _BuildInstance._invokeInternal (package:flutter_tools/src/build_system/build_system.dart:526:25)
    <asynchronous suspension>
    #2      _BuildInstance.invokeTarget.<anonymous closure> (package:flutter_tools/src/build_system/build_system.dart:481:35)
    #3      new Future.sync (dart:async/future.dart:224:31)
    #4      AsyncMemoizer.runOnce (package:async/src/async_memoizer.dart:43:45)
    #5      _BuildInstance.invokeTarget (package:flutter_tools/src/build_system/build_system.dart:481:21)
    <asynchronous suspension>
    <asynchronous suspension>
    #6      BuildSystem.build (package:flutter_tools/src/build_system/build_system.dart:419:36)
    #7      _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:43:6)
    #8      BuildSystem.build (package:flutter_tools/src/build_system/build_system.dart:400:28)
    #9      buildWithAssemble (package:flutter_tools/src/bundle.dart:172:48)
    #10     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:43:6)
    #11     buildWithAssemble (package:flutter_tools/src/bundle.dart:148:31)
    #12     BundleBuilder.build (package:flutter_tools/src/bundle.dart:85:13)
    #13     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:43:6)
    #14     BundleBuilder.build (package:flutter_tools/src/bundle.dart:58:21)
    #15     BuildBundleCommand.runCommand (package:flutter_tools/src/commands/build_bundle.dart:126:25)
    #16     _AsyncAwaitCompleter.start (dart:async-patch/async_patch.dart:43:6)
    #17     BuildBundleCommand.runCommand (package:flutter_tools/src/commands/build_bundle.dart:97:42)
    #18     FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:557:18)
    #19     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
    #20     _rootRunUnary (dart:async/zone.dart:1132:38)
    #21     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
    #22     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
    #23     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
    #24     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
    #25     Future._completeWithValue (dart:async/future_impl.dart:522:5)
    #26     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:552:7)
    #27     _rootRun (dart:async/zone.dart:1124:13)
    #28     _CustomZone.run (dart:async/zone.dart:1021:19)
    #29     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
    #30     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
    #31     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
    #32     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
    #33     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
    #34     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)

    Failed to build bundle.
    Failed to package /Users/tom/Development/skp-client/teambeam.
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build systemnote: Planning buildnote: Constructing build description

Could not build the precompiled application for the device.
```

**Comment by Sh1d0w** [comment]
This plugin supports only the stable channel. There is no way to support unstable stuff on the other channels and the stable channel.

**Comment by technolion** [comment]
OK, then this will pop up again later, when the commit on flutter master channel will be merged to stable. Since I am writing a Flutter desktop app I am forced to use the master branch.

Your your info, I believe this commit caused the breakage:
https://github.com/flutter/flutter/commit/b85763230658c2eea18f541715deb068fabc0b7b

**Comment by georgeherby** [comment]
Now that this is on master as per #287 should this not reopened?

**Comment by Sh1d0w** [comment]
@georgeherby Flutter master channel is unstable. The plugin strives to support only the stable channel of Flutter.

Once their master is merged into stable I will implement a fix, until then this is a breaking change that I can't support.

I already had similar issue months ago again in with Flutter's master channel. I've implemented a fix, but several days ago Flutter team decided to rollback their commit, so did I. It is not practical to support unstable stuff.

I would advise you to always use stable channel, as breakages will occur frequently.

If for some reason you need to use master, feel free to fork the repo and implement the fix for yourself until this is released into stable.

**Comment by georgeherby** [comment]
@Sh1d0w Understood - always good when their master branch is not stable...I have raised a PR with the fix if people want to use it :) As I updated to the new version of master and now my projects won't build :)

If anyone wants to use the fork
```
  multi_image_picker:
    git:
      url: git://github.com/georgeherby/multi_image_picker.git
      ref: image-stream-completer-fix
```

**Comment by ThinkDigitalSoftware** [comment]
> This plugin supports only Flutter on the stable channel. There is no way to support unstable stuff on the other channels and the stable channel.

You could do what other libraries do and release 2 versions. For example, [cached_network_image](https://pub.dev/packages/cached_network_image) has 2 versions that default to the stable version, but allows for those who use `master` to have a version as well.
It's a perfect solution IMO. 

<img width="473" alt="Screen Shot 2019-10-29 at 12 29 22 PM" src="https://user-images.githubusercontent.com/23037821/67802226-f74ab080-fa47-11e9-8405-472c940d3f8e.png">

**Comment by Sh1d0w** [comment]
Hi, this is now fixed in version 4.6.0-rc.1

**Comment by ThinkDigitalSoftware** [comment]
Yo, thank you!

**Comment by jonathanots** [comment]
> @Sh1d0w Understood - always good when their master branch is not stable...I have raised a PR with the fix if people want to use it :) As I updated to the new version of master and now my projects won't build :)
> 
> If anyone wants to use the fork
> 
> ```
>   multi_image_picker:
>     git:
>       url: git://github.com/georgeherby/multi_image_picker.git
>       ref: image-stream-completer-fix
> ```

Thank you. It's working fine.
